punchcut/docs/capi.md

```diff
@@ -0,0 +1,154 @@
+# C API for punchcut
+
+This is a rough proposal for a C API for punchcut.
+
+## Header
+
+```c
+#include <stdint.h>
+#include <stddef.h>
+
+struct PcContext;
+struct PcScalerBuilder;
+struct PcScaler;
+struct PcOutline;
+
+// Supported hinting modes.
+enum PcHinting {
+    // No hinting.
+    PC_HINTING_NONE = 0,
+    // Full gridfitting.
+    PC_HINTING_LEGACY = 1,
+    // Grayscale antialiasing (FreeType light + grayscale)
+    PC_HINTING_GRAYSCALE_SUBPIXEL = 2,
+    // RGB subpixel hinting (FreeType light + LCD)
+    PC_HINTING_SUBPIXEL = 3,
+    // Same as RGB subpixel, but prevents all movement
+    // in the horizontal direction.
+    PC_HINTING_MODERN = 4,
```

Comment:
is this name in general use? I have a slight allergy to the use of 'modern' in general, since it doesn't really communicate much more than "new when I named this".

Reply:
"Linear" might describe it better. Also, what's up with vertical text hinting, anyway?

Reply:
Also, the options here seem a bit arbitrary... Gray-scale/Subpixel and amount and direction of gridfitting are orthogonal concepts, where for the latter we have accidental control in TTFs (hinting done by code in font) and full control in CFFs (hinting done by renderer).

Reply:
The grayscale/subpixel options are not necessarily arbitrary. These are returned as flags in the GETINFO instruction and do actually affect hinting, specifically for fonts in the ClearType collection. The names are unfortunate though.

I suggest replacing them with Full, Light, LightSubpixel and VerticalSubpixel. Do those make more sense?

Reply:
These seem like improvements over "legacy" and "modern". As we can't find the full technical description into the enum value, these aliases or shorthands work for me.